src/zarr/registry.py

```diff
@@ -42,28 +51,52 @@ def __init__(self) -> None:
 
     def lazy_load(self) -> None:
         for e in self.lazy_load_list:
-            self.register(e.load())
+            cls = e.load()
+
+            if (hasattr(cls, "_zarr_spec_identifier") and hasattr(cls, "to_numpy")):
```

Comment:
The motivation here is that I don't want a user to be able to add a dtype whose `numpy` representation overrides an already registered dtype e.g. MyFakeBool should not be able to override the core Bool dtype.

The exceptions are for the V and B dtypes, because these are raw byte strings, and multiple dtypes might be mapped to this e.g. `int2` and `int4` might both be mapped to the 'B' dtype in `numpy` because `numpy` has no support for sub-byte dtypes.